kie-wb-common-react/boxed-expression-editor/src/hooks/Hooks.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Dispatch, RefObject, SetStateAction, useCallback, useEffect, useRef, useState } from "react";
+
+export function useContextMenuHandler(): [
+  RefObject<HTMLDivElement>,
+  string,
+  string,
+  boolean,
+  Dispatch<SetStateAction<boolean>>
+] {
+  const wrapperRef = useRef<HTMLDivElement>(null);
+
+  const [xPos, setXPos] = useState("0px");
+  const [yPos, setYPos] = useState("0px");
+  const [contextMenuIsVisible, setContextMenuVisibility] = useState(false);
+
+  const hideContextMenu = useCallback(() => {
+    contextMenuIsVisible && setContextMenuVisibility(false);
+  }, [contextMenuIsVisible]);
+
+  const showContextMenu = useCallback(
+    (event: MouseEvent) => {
+      if (wrapperRef.current && wrapperRef.current === event.target) {
+        event.preventDefault();
+        setXPos(`${event.pageX}px`);
+        setYPos(`${event.pageY}px`);
+        setContextMenuVisibility(true);
+      }
+    },
+    [setXPos, setYPos]
+  );
+
+  useEffect(() => {
+    document.addEventListener("click", hideContextMenu);
+    document.addEventListener("contextmenu", showContextMenu);
+    return () => {
+      document.removeEventListener("click", hideContextMenu);
+      document.removeEventListener("contextmenu", showContextMenu);
+    };
+  });
+
+  return [wrapperRef, xPos, yPos, contextMenuIsVisible, setContextMenuVisibility];
```

Comment:
Nice point. I was in doubt. I just used the same approach of out-of-the-box hooks, like `useState` which is returning an array.

JavaScript array destructuring syntax allows you to name each returned element.
When you return an object, you are forced to either use the object property name or need to alias it.

Reply:
I think in this case an object could be better because the returned variables are always the same. I don't see a real need for an array. It's different for `useState` because you actually need to specify different names every time you use it.

Reply:
Ok pushed the change!